app/components/InputSelect.tsx

```diff
@@ -35,6 +36,8 @@ export type Props = {
   options: Option[];
   note?: React.ReactNode;
   onChange: (value: string | null) => void;
+  renderOption?: (option: Option, select: SelectStateReturn) => React.ReactNode;
```

Comment:
I don't understand why these changes are needed, the idea is that you can pass the rendered items directly into the `options[]` prop as elsewhere so custom render methods aren't required.

One pattern we've used is to just create a parent component per-select-type, so in this case there might be a `DefaultCollectionSelectInput` for example that is capable of loading the data it needs and passing the correct values to the `options` prop.

Does that make sense?